WordPress/src/main/java/org/wordpress/android/viewmodel/posts/PostListItemUiStateHelper.kt

```diff
@@ -394,14 +396,11 @@ class PostListItemUiStateHelper @Inject constructor(
 
         if (canShowPublishButton) {
             buttonTypes.add(
-                    if (canRetryUpload) {
-                        BUTTON_RETRY
-                    } else if (!siteHasCapabilitiesToPublish) {
-                        BUTTON_SUBMIT
-                    } else if (postStatus == SCHEDULED && isLocallyChanged) {
-                        BUTTON_SYNC
-                    } else {
-                        BUTTON_PUBLISH
+                    when {
+                        canRetryUpload -> BUTTON_RETRY
+                        !siteHasCapabilitiesToPublish -> BUTTON_SUBMIT
+                        postStatus == SCHEDULED && isLocallyChanged -> BUTTON_SYNC
+                        else -> BUTTON_PUBLISH
```

Comment:
Thank you for improving this part of the code 🙇

Reply:
My pleasure! 😇